File: src/main/java/cloud/JsonUtil.java
package cloud;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;

import com.google.gson.Gson;

public final class JsonUtil {

	private static Gson gson = new Gson();

	public static <T> T fromJsonStream(InputStream is, Class<T> classOfT) {
		return fromJsonReader(new BufferedReader(new InputStreamReader(is)), classOfT);
	}

	public static <T> T fromJsonReader(BufferedReader reader, Class<T> classOfT) {
		try {
			StringBuilder sb = new StringBuilder();
			String line = "";
			while ((line = reader.readLine()) != null) {
				sb.append(line);
			}
			return gson.fromJson(sb.toString(), classOfT);
		} catch (IOException e) {
			e.printStackTrace();
			return null;
		}
	}
	
	public static String toJson(Object src) {
		return gson.toJson(src);
	}
	
	
	public  static <T> T fromJson(String json,Class<T> classOfT) {
		return gson.fromJson(json,classOfT);
	}
}

File: src/main/kotlin/cloud/util/JsonUtil.kt
package cloud.util

import com.google.gson.Gson
import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader

object JsonUtil {

    val gson = Gson()

    fun <T> fromJsonStream(input: InputStream, classOfT: Class<T>): T {
        return fromJsonReader(BufferedReader(InputStreamReader(input)), classOfT)
    }

    fun <T> fromJsonReader(reader: BufferedReader, classOfT: Class<T>): T {

        val sb = StringBuilder()
        var line: String? = ""
        while (reader.readLine().also { line = it } != null) {
            sb.append(line)
        }
        return gson.fromJson(sb.toString(), classOfT)
    }

    fun toJson(src: Any?): String {
        return gson.toJson(src)
    }

    fun <T> fromJson(json: String?, classOfT: Class<T>?): T {
        return gson.fromJson(json, classOfT)
    }
}